### PR TITLE
Add transformation tools

### PR DESCRIPTION
**Pull request by eric-wieser**
These will help with basis changes, most notably handy for CGA.

~~Hopefully the doctests are enough to produce coverage data...~~

With thanks to @enki and @jeremyong for helping me work this out, and @brombo for a very similar class in `galgebra`. Naming suggestions would be appreciated.

### CODE REVIEW

File: clifford/_transformation.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+        # fill the scalar
+        self[(),()] = 1
+
+        # fill the vectors
+        for i_from in range(layout_from.dims):
+            vector_from = (i_from+layout_from.firstIdx,)
+            for i_to in range(layout_to.dims):
+                vector_to = (i_to+layout_to.firstIdx,)
+                self[vector_to, vector_from] = matrix[i_to, i_from]
```
Comment (eric-wieser) [inline clifford/_transformation.py]: Assigning by tuple like this is probably super slow, but optimizing it is a task for later - preferably once we can guarantee the blade order in the `Layout` class.

**Comment by eric-wieser** [comment]
Will come back to this shortly and add `LinearTransformation.from_function(lambda x: ...)`, which can be used as something like:
```
def rotor_to_matrix(R):
    return LinearTransformation.from_function(lambda x: R * x * ~R).vector_matrix()
```

File: clifford/_transformation.py (hunk 1 above)
Comment (eric-wieser) [reply clifford/_transformation.py]: The changes above have been merged since I made this PR - the latest revision makes use of the new features.

**Comment by hugohadfield** [comment]
This looks nice, should hopefully allow us to make some nice tools for swapping between (sub)algebras easily and efficiently as well as swallowing up things like:
https://github.com/pygae/clifford/blob/93b0188652e96c7768a658abea4ffccbb8fc358b/clifford/__init__.py#L107 
 and other assorted linear algebra gubbins in the tools section. 

My only slight concern with it is the name, transformations for me in GA screams rotors and bivectors rather than linear algebra. What would you think about clifford.linalg ?

File: clifford/transformations.py [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+    def adjoint(self) -> "LinearMatrix":
+        """ The adjoint transformation """
+        return LinearMatrix(self._matrix.T, self.layout_dst, self.layout_src)
+
+
+class OutermorphismMatrix(LinearMatrix):
+    r"""
+    A generalization of a linear transformation to vectors via the outer product.
+
+    Namely, given a linear transformation :math:`F(u) \to v`, this generalize
```
Comment (hugohadfield) [inline clifford/transformations.py:222]: this generalize to

Comment (hugohadfield) [reply clifford/transformations.py:222]: should be 'generalizes'

**Comment by eric-wieser** [comment]
Finally got the link to the apollonius tutorial to work, will merge once CI is done.

**Comment by eric-wieser** [comment]
CI failure is the numba RC issue again, so out of scope.

**Pull request merged**